internal/common/common.go

```diff
@@ -73,6 +73,87 @@ type ReferenceValues struct {
 	BuilderImageSHA256Digests []string `toml:"builder_image_sha256_digests"`
 }
 
+// ProvenanceIR is an internal intermediate representation of data from provenances for verification.
+// We use the ProvenanceIR to map different provenance formats.
+type ProvenanceIR struct {
+	binarySHA256Digest       string
+	buildCmd                 []string
+	builderImageSHA256Digest string
+}
+
+type ProvenanceIROption = func(p *ProvenanceIR)
+
+func NewProvenanceIR(binarySHA256Digest string, options ...func(*ProvenanceIR)) *ProvenanceIR {
```

Comment:
Does it work?

```suggestion
func NewProvenanceIR(binarySHA256Digest string, options ... ProvenanceIROption) *ProvenanceIR {
```

Reply:
Yes, it does! Now we can either keep the type `ProvenanceIROption` or replace it with` func(p *ProvenanceIR)` everywhere. I prefer the latter, but both are fine for me.